src/pages/index.js

```diff
@@ -43,83 +116,134 @@ profileEditButton.addEventListener("click", () => {
+  addCardFormValidator.toggleButtonState();
+  addCardModal.open();
+});
+
+avatarPictureButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+  avatarFormValidator.toggleButtonState();
+});
 
-const editProfileModal = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
 
 /* ----------------------- */
-/*     ADD CARD            */
+/*     function            */
 /* ----------------------- */
 
-function handleAddCardFormSubmit(formData) {
-  const card = getCard({ name: formData.name, link: formData.url });
-  cardSection.addItem(card);
-  addCardModal.close();
-}
 
-function getCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImagePreview); //call out Card class with corresponding argument in Card class constructor
-  return cardElement.getNewCard();
+function handleProfileEditSubmit({name, description}) {//name and description have to match the name of profile edit form input name
+  editProfileModal.setLoading(true, "Saving...");
+  api.patchProfile(name, description).then((userData) => {
+    console.log(userData);
+    profileUserInfo.setUserInfo(userData.name, userData.about);
+    editProfileModal.close();
+  }).finally(() => { 
+    editProfileModal.setLoading(false);
+  })
+  .catch((err)=> {
+    console.error(err);
+  })
 }
 
-const createCard = (item) => {
-  const newCard = getCard(item);
-  cardSection.addItem(newCard);
-};
+function handleAddCardFormSubmit({name, url}) {
+  addCardModal.setLoading(true,"Saving...");
+  api.postCards(name, url).then((cardData) => {
+    // console.log(cardData);
+    const card = getCard(cardData);
+    // addCardFormValidator.toggleButtonState();
+    cardSection.addItem(card);
+    addCardModal.close();
+
+  }).finally(() => {
+    addCardModal.setLoading(false);
+  }).catch((err) => {
+    console.error(err);
+  })
 
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
-);
+}
 
-addNewCardButton.addEventListener("click", () => {
-  addCardFormValidator.toggleButtonState();
-  addCardModal.open();
-});
+// function getCard (cardData) {
+//   const cardElement = new Card(cardData, "#card-template", handleImagePreview); //call out Card class with corresponding argument in Card class constructor
+//   return cardElement.getNewCard();
+// }
+
+ 
+function getCard(items) {
+  const cardElement = new Card(
+    items,
+    "#card-template",
+    handleImagePreview,
+    handleCardDeleteSubmit,
+    handleLikeClick
+  );
 
-const addCardFormValidator = new FormValidator(addCardEditForm, options);
+  return cardElement.getNewCard();
+}
 
-const addCardModal = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
-);
 
-/* ----------------------- */
-/*     Image Preview      */
-/* ----------------------*/
+function handleLikeClick(items) {
+  //toggle the like status
+  api.cardLikeStatus(items._id,items._isLiked).then(() => {
+    items.updateLikes(items._isLiked);
```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.